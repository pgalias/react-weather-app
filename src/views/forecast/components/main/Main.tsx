import React, { FunctionComponent } from 'react';
import { City } from '../../../../models';

import styles from './main.module.css';

type Props = {
  airTemperature: number;
  airTemperatureUnit: string;
  city: City;
  className: string;
};

export const Main: FunctionComponent<Props> = ({
  airTemperature,
  airTemperatureUnit,
  city,
  className,
}) => {
  const unit = airTemperatureUnit[0];

  return (
    <main className={`${styles.main} ${className}`}>
      <p className={styles.temperature}>
        {airTemperature} Â°{unit.toUpperCase()}
      </p>
      <p className={styles.city}>
        {city.city}, {city.state}, {city.country}
      </p>
    </main>
  );
};
