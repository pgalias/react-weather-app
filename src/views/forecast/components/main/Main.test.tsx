import React from 'react';
import { render, screen } from '@testing-library/react';
import { Main } from './Main';

describe('forecast::components::Main', () => {
  test('should render passed air temperature and city info', () => {
    render(
      <Main
        airTemperature={20}
        airTemperatureUnit="celsius"
        city={{ city: 'New York City', state: 'New York', country: 'US' }}
        className="foo"
      />,
    );

    expect(screen.getByRole('main')).toMatchSnapshot();
    expect(screen.getByText(/20 Â°C/)).toBeInTheDocument();
    expect(screen.getByText(/New York City, New York, US/)).toBeInTheDocument();
  });
});
