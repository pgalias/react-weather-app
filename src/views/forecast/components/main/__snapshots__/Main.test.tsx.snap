// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forecast::components::Main should render passed air temperature and city info 1`] = `
<main
  class="main foo"
>
  <p
    class="temperature"
  >
    20
     Â°
    C
  </p>
  <p
    class="city"
  >
    New York City
    , 
    New York
    , 
    US
  </p>
</main>
`;
